pkg/supervisor/archived_snapshot.go

```diff
@@ -97,42 +107,83 @@ func (c *s3Snapshot) Upload(ctx context.Context, path string) error {
 	return nil
 }
 
+func getCheckSum(path string) (string, error) {
+	f, err := os.OpenFile(path, os.O_RDONLY, 0)
+	if err != nil {
+		return "", errors.Wrap(err, "opening file")
+	}
+	defer f.Close()
+
+	h := sha256.New()
+	if _, err := io.Copy(h, f); err != nil {
+		events.Log("failed to generate sha256", err)
+	}
+
+	cs := base64.StdEncoding.EncodeToString(h.Sum(nil))
+	events.Log("base64 encoding: %s", cs)
+
+	return cs, nil
+}
+
 func isCompressed(gpr *gzipCompressionReader) string {
 	if gpr == nil {
 		return "false"
 	}
 	return "true"
 }
 
-func (c *s3Snapshot) sendToS3(ctx context.Context, key string, bucket string, body io.Reader) error {
+type BucketBasics struct {
+	S3Client S3Client
+}
+
+func (c *s3Snapshot) sendToS3(ctx context.Context, key string, bucket string, body io.Reader, cs string) error {
 	if c.sendToS3Func != nil {
 		return c.sendToS3Func(ctx, key, bucket, body)
 	}
-	ul, err := c.getS3Uploader()
+
+	client, err := c.getS3Client()
 	if err != nil {
 		return err
 	}
-	output, err := ul.UploadWithContext(ctx, &s3manager.UploadInput{
-		Bucket: &bucket,
-		Key:    &key,
-		Body:   body,
+
+	var basics = BucketBasics{
+		S3Client: client,
+	}
+	var partMiBs int64 = 16
+	uploader := manager.NewUploader(basics.S3Client, func(u *manager.Uploader) {
+		u.PartSize = partMiBs * 1024 * 1024
+	})
+
+	output, err := uploader.Upload(ctx, &s3.PutObjectInput{
+		Bucket:            &bucket,
+		Key:               &key,
+		Body:              body,
+		ChecksumAlgorithm: "sha256",
+		Metadata: map[string]string{
```

Comment:
The reason why I am using a general purpose `Metadata` rather than AWS official doc recommended `ChecksumSHA256` is because 
<img width="706" alt="Screenshot 2023-09-26 at 5 07 38 PM" src="https://github.com/segmentio/ctlstore/assets/6007569/13702b33-706e-44ae-8222-ae71109495a5">
It would be over complicated if we do part level checksum calculation & validation.